src/langchain_google_cloud_sql_mysql/mysql_engine.py

```
@@ -109,34 +117,56 @@ def from_instance(
+            user=user,
+            password=password,
         )
         return cls(engine=engine)
 
     @classmethod
     def _create_connector_engine(
-        cls, instance_connection_name: str, database: str
+        cls,
+        instance_connection_name: str,
+        database: str,
+        user: Optional[str],
+        password: Optional[str],
     ) -> sqlalchemy.engine.Engine:
         """Create a SQLAlchemy engine using the Cloud SQL Python Connector.
 
         Defaults to use "pymysql" driver and to connect using automatic IAM
         database authentication with the IAM principal associated with the
-        environment's Google Application Default Credentials.
+        environment's Google Application Default Credentials. If user and
+        password arguments are given, basic database authentication will be
+        used for database login.
 
         Args:
             instance_connection_name (str): The instance connection
                 name of the Cloud SQL instance to establish a connection to.
                 (ex. "project-id:instance-region:instance-name")
             database (str): The name of the database to connect to on the
                 Cloud SQL instance.
+            user (str, optional): Database user to use for basic database
+                authentication and login. Defaults to None.
+            password (str, optional): Database password for 'user' to use for
+                basic database authentication and login. Defaults to None.
+
         Returns:
             (sqlalchemy.engine.Engine): Engine configured using the Cloud SQL
                 Python Connector.
         """
-        # get application default credentials
-        credentials, _ = google.auth.default(
-            scopes=["https://www.googleapis.com/auth/userinfo.email"]
-        )
-        iam_database_user = _get_iam_principal_email(credentials)
+        # if user and password are given, use basic auth
+        if user and password:
```

should we throw an error if they specify user and not password (or vica versa)?

---

added an error, catching it up a level in the `.from_instance()` class method for easier developer experience with stacktrace coming from a public method instead of the internal `_create_connector_engine`